tmt/steps/prepare/install.py

```diff
@@ -586,24 +555,54 @@ def go(
             return
 
         # Pick the right implementation
-        # TODO: it'd be nice to use a "plugin registry" and make the implementations
-        # discovered as any other plugins.
-        if guest.facts.package_manager == GuestPackageManager.RPM_OSTREE:
-            installer: InstallBase = InstallRpmOstree(logger=logger, parent=self, guest=guest)
-
-        elif guest.facts.package_manager == GuestPackageManager.DNF:
-            installer = InstallDnf(logger=logger, parent=self, guest=guest)
-
-        elif guest.facts.package_manager == GuestPackageManager.DNF5:
-            installer = InstallDnf5(logger=logger, parent=self, guest=guest)
-
-        elif guest.facts.package_manager == GuestPackageManager.YUM:
-            installer = InstallYum(logger=logger, parent=self, guest=guest)
+        # TODO: it'd be nice to use a "plugin registry" and make the
+        # implementations discovered as any other plugins. Package managers are
+        # shipped as plugins, but we still need a matching *installation* class.
+        # But do we really need a class per package manager family? Maybe the
+        # code could be integrated into package manager plugins directly.
+        if guest.facts.package_manager == 'rpm-ostree':
+            installer: InstallBase = InstallRpmOstree(
```

Comment:
Do we need these 'InstallBase'? Can it be part of package_manager API? Guest already knows which package manager to use, why should prepare install have this switch?

Reply:
It's likely the answer is "no, we don't need them". I tried to extract some basic functionality, and I might have ended up with leftovers in these classes that will not survive in the long run. I was hoping to sync on the high-level behavior and handling with @thrix while he was working on the all-mighty install plugin. I was certainly expecting we'd get rid of them some day, maybe we should do so sooner.

Reply:
My approach here was to separate the primitives like "install a package" into package manager classes, while the `InstallBase` classes would handle higher level - which packages to install together, separating debuginfo, deciding whether installing all packages at once is the right way or not, enabling Copr, etc. I suppose this can get into package managers too, but my first patch tried to focus on implementing the very basic actions only. It's a dangerous area, and it felt like the minimal viable change I could start with without closing the door for future changes.

Reply:
I see. Let's keep it there for now.